configure.ac

```diff
@@ -839,27 +839,17 @@ AC_ARG_ENABLE(esi,
                  [squid_opt_use_esi=$enableval],[])
 
 # ESI support libraries: expat
-AH_TEMPLATE(HAVE_LIBEXPAT,[Define to 1 if you have the expat library])
 SQUID_AUTO_LIB(expat,[ESI expat library],[LIBEXPAT])
-AS_IF([test "x$squid_opt_use_esi" != "xno" -a "x$with_expat" != "xno"],[
-  SQUID_STATE_SAVE(squid_expat_state)
-  PKG_CHECK_MODULES([LIBEXPAT],[expat],[],[
-    AC_CHECK_LIB([expat],[main],[LIBEXPAT_LIBS="-lexpat"])
+SQUID_CHECK_LIB_WORKS(expat,[
+  AS_IF([test "x$squid_opt_use_esi" != "xno"],[
+    SQUID_STATE_SAVE(squid_expat_state)
+    PKG_CHECK_MODULES([LIBEXPAT],[expat],[],[])
+    CPPFLAGS="$LIBEXPAT_CFLAGS $CPPFLAGS"
+    AC_CHECK_HEADERS(expat.h)
+    SQUID_STATE_ROLLBACK(squid_expat_state)
   ])
-  CPPFLAGS="$LIBEXPAT_CFLAGS $CPPFLAGS"
-  AC_CHECK_HEADERS(expat.h)
-  SQUID_STATE_ROLLBACK(squid_expat_state)
-  AS_IF([test "x$LIBEXPAT_LIBS" != "x"],[
-      squid_opt_use_esi=yes
-      LIBEXPAT_LIBS="$LIBEXPAT_PATH $LIBEXPAT_LIBS"
-      AC_DEFINE(HAVE_LIBEXPAT,1,[Define to 1 if you have the expat library])
-    ],
-    [test "x$with_expat" = "xyes"],[AC_MSG_ERROR([Required library expat not found.])],
-    [AC_MSG_NOTICE([Library expat not found.])]
-  )
 ])
-AM_CONDITIONAL(ENABLE_LIBEXPAT,[test "x$LIBEXPAT_LIBS" != "x"])
-AC_SUBST(LIBEXPAT_LIBS)
+AS_IF([test "x$with_expat" = "xyes"],[squid_opt_use_esi=yes])
```

Comment:
The new check macro explicitly sets `with_foo` to "no" unless the library is both found and working.

Since I have expanded scope to include libxml2 and simplified the final `AS_IF` now this `squid_opt_use_esi` has gone away.